models/transaction_validation.go

```diff
@@ -0,0 +1,65 @@
+package models
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/onflow/go-ethereum/common"
+	"github.com/onflow/go-ethereum/core/types"
+)
+
+func ValidateTransaction(tx *types.Transaction) error {
```

Comment:
It would be better to make the function be part of the model, so inside models/transactions you could do this on:

```
func (t *types.Transaction) Validate() error
```

Reply:
ohh right, my bad, just another nit is can we move this in the transaction.go, it's not a pattern to have a file named by the function, otherwise approved

Reply:
Good idea :ok_hand: 
Moved in https://github.com/onflow/flow-evm-gateway/pull/314/commits/dd5495f125ec1bfa966d1505b686636a72475439